gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegionMap.java

```diff
@@ -3864,12 +3864,17 @@ public RegionEntry basicPut(EntryEventImpl event,
       // wait for the initialization to get over which will not finish
       // until this testLatch is counted down from the test code
       if (DistributedRegion.testLatch == null) {
```

Comment:
We can still keep the !owner.initialized() condition as acquiring the readLock is not necessary when the region is initialized.

Reply:
There is a check just above for the same. So, if region is initialized, this code path will not be executed.

Reply:
Cool. The context was not that hide to include that statement so missed it.